src/connectionhealthpolicy/SendingAudioFailureConnectionHealthPolicy.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import BaseConnectionHealthPolicy from './BaseConnectionHealthPolicy';
+import ConnectionHealthData from './ConnectionHealthData';
+import ConnectionHealthPolicy from './ConnectionHealthPolicy';
+import ConnectionHealthPolicyConfiguration from './ConnectionHealthPolicyConfiguration';
+
+export default class SendingAudioFailureConnectionHealthPolicy
+  extends BaseConnectionHealthPolicy
+  implements ConnectionHealthPolicy {
+  private readonly secondsToConsider: number;
+  private readonly initialWaitTimeMs: number;
+  private readonly coolDownTimeMs: number;
+  private readonly maximumTimesToWarn: number;
+  private warnCount: number;
+  private lastWarnTimestampMs: number;
+
+  constructor(
+    private logger: Logger,
+    configuration: ConnectionHealthPolicyConfiguration,
+    data: ConnectionHealthData
+  ) {
+    super(configuration, data, 'Sending Audio Health');
+    this.secondsToConsider = configuration.sendingAudioFailureSamplesToConsider;
+    this.initialWaitTimeMs = configuration.sendingAudioFailureInitialWaitTimeMs;
+    this.maximumTimesToWarn = configuration.maximumTimesToWarn;
+    this.coolDownTimeMs = configuration.cooldownTimeMs;
+    this.lastWarnTimestampMs = 0;
+    this.warnCount = 0;
+  }
+
+  private isSendingAudioUnHealthy(): boolean {
+    const hasEnoughTimeElapsedToEvaluateStatus = !this.currentData.isConnectionStartRecent(
+      this.initialWaitTimeMs
+    );
+    const areAudioPacketsNotBeingSent =
+      this.currentData.consecutiveStatsWithNoAudioPacketsSent >= this.secondsToConsider;
+    return hasEnoughTimeElapsedToEvaluateStatus && areAudioPacketsNotBeingSent;
+  }
+
+  health(): number {
+    const didWarnRecently = Date.now() - this.lastWarnTimestampMs < this.coolDownTimeMs;
+    if (this.isSendingAudioUnHealthy()) {
+      if (this.currentHealth > this.minimumHealth() && !didWarnRecently) {
+        this.logger.debug(`Sending Audio is unhealthy for ${this.secondsToConsider} seconds`);
+        this.warnCount++;
+        if (this.warnCount > this.maximumTimesToWarn) {
+          this.logger.debug(
```

Comment:
should this be debug level?

Reply:
Was confused because out of the two existing health policies one has logging at warn level and the other has no logging at all. I'll change this to warn as it provides better visibility and there's not a lot of performance impact either (2 logs every 1 min in worst case)